pkg/controller/worker/helper.go

```diff
@@ -58,3 +64,44 @@ func (w *workerDelegate) updateWorkerProviderStatus(ctx context.Context, workerS
 		return nil
 	})
 }
+
+func (w *workerDelegate) decodeWorkerPoolProviderConfig(pool extensionsv1alpha1.WorkerPool) (*api.WorkerPoolProviderConfig, error) {
+	if pool.ProviderConfig == nil || pool.ProviderConfig.Raw == nil {
+		return nil, nil
+	}
+
+	pc, err := helper.WorkerPoolProviderConfigFromRawExtension(pool.ProviderConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not decode WorkerPoolProviderConfig '%s/%s'", kutil.ObjectName(w.worker), pool.Name)
+	}
+
+	return pc, nil
+}
+
+func (w *workerDelegate) updateMachineDependenciesStatus(ctx context.Context, workerStatus *api.WorkerStatus, ServerGroupDependencies []api.ServerGroupDependency, err error) error {
+	workerStatus.ServerGroupDependencies = ServerGroupDependencies
+	if statusUpdateErr := w.updateWorkerProviderStatus(ctx, workerStatus); statusUpdateErr != nil {
+		if err != nil {
+			err = errors.Wrapf(statusUpdateErr, err.Error())
+		} else {
+			err = statusUpdateErr
+		}
+	}
+
+	return err
+}
+
+// ClusterTechnicalName returns the technical name of the cluster this worker manages
+func (w *workerDelegate) ClusterTechnicalName() string {
```

Comment:
Why do we need a function for this?

Reply:
Arguably unnecessary. 

Sometimes I like to use functions as a small wrapper of context, in this case conveying the concept that the names of the resources we create have a special convention and used for pseudo-labeling. Otherwise someone that reads the code for the first time could find it confusing why there is a reference to the `w.cluster.objectMeta.name` in a context that it is not directly related to the cluster resource.

Reply:
OK, good argument, thanks!